src/ImageSharp/Color/Color.cs

```diff
@@ -91,7 +96,18 @@ private Color(byte r, byte g, byte b)
         /// <param name="b">The blue component (0-255).</param>
         /// <returns>The <see cref="Color"/>.</returns>
         [MethodImpl(InliningOptions.ShortMethod)]
-        public static Color FromRgb(byte r, byte g, byte b) => new Color(r, g, b);
+        public static Color FromRgb(byte r, byte g, byte b) => new(r, g, b);
+
+        /// <summary>
+        /// Creates a <see cref="Color"/> from the given <typeparamref name="TPixel"/>.
+        /// </summary>
+        /// <param name="pixel">The pixel to convert from.</param>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <returns>The <see cref="Color"/>.</returns>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public static Color FromPixel<TPixel>(TPixel pixel)
+            where TPixel : unmanaged, IPixel<TPixel>
+            => new(pixel);
```

Comment:
I'm surprised we didn't have this method before. We should avoid boxing in case user provides a typical pixel like `Rgba32`, which I would expect to be the more common than advanced scenarios like https://github.com/SixLabors/ImageSharp.Drawing/discussions/165

I realized that we don't need to add new stuff to `PixelOperations<T>` unless we see users complaining about the default heuristic, which is very unlikely.

```suggestion
            => Unsafe.SizeOf<TPixel> > sizeof(Rgba64)  ? new(pixel) : new(pixel.ToRgba64());
```

Reply:
Does sizeof work here? Does the compiler see that as a constant?

Reply:
>  Does the compiler see that as a constant?

As far as I'm aware it should, since it's not a generic argument, however we need to mark the method unsafe too which is not in my recommendation.

Reply:
Just remembered why I wrote it this way. `ToRgba64` doesn't exist on `IPixel<TSelf>`;

Reply:
> `ToRgba64` doesn't exist on `IPixel<TSelf>`;

I missed that. We can implement this as following:

```C#
[MethodImpl(InliningOptions.ShortMethod)]
public unsafe static Color FromPixel<TPixel>(TPixel pixel)
    where TPixel : unmanaged, IPixel<TPixel>
{
    // Avoid boxing in case we can convert to Rgba64 safely and efficently
    if (typeof(TPixel) == typeof(Rgba64))
    {
        Rgba64 p = (Rgba64)(object)pixel;
        return new(p);
    }
    else if (typeof(TPixel) == typeof(Rgb48))
    {
        Rgb48 p = (Rgb48)(object)pixel;
        return new(new Rgba64(p.R, p.G, p.B, ushort.MaxValue)); // or add a new Color(Rgb48) constructor
    }
    else if (Unsafe.SizeOf(TPixel) <= sizeof(Rgba32))
    {
        return new(pixel.ToRgba32());
    }
    else
    {
        return new(pixel);
    }
}
```

The JIT will optimize away the unnecessary branches and the boxing from the `(Rgba64)(object)pixel` conversion.

Reply:
Cool, will do.

Reply:
La32 also.